java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -245,6 +245,9 @@ protected final void serializeSomethingField(
             StructBondType.StructField<T> field) throws IOException {
         if (value != null) {
             serializeField(context, value.getValue(), field);
+        } else if (!field.isOptional()) {
+            // throws
```

Comment:
Minor: I find the `Throw.raise...` line to be clear enough on its own, so I'd remove the comment.